src/pokemon.c

```diff
@@ -6941,3 +6942,150 @@ void UpdateDaysPassedSinceFormChange(u16 days)
         }
     }
 }
+
+u8 CheckDynamicMoveType(struct Pokemon *mon, u16 move, u32 battler)
+{
+    u8 type = gMovesInfo[move].type;
+    u16 species = GetMonData(mon, MON_DATA_SPECIES);
+    u16 heldItem = GetMonData(mon, MON_DATA_HELD_ITEM, 0);
+    u16 heldItemEffect = ItemId_GetHoldEffect(heldItem);
+    u16 ability = GetMonAbility(mon);
+    u8 type1 = gSpeciesInfo[species].types[0];
+    u8 type2 = gSpeciesInfo[species].types[1];
+
+
+    if (P_SHOW_DYNAMIC_TYPES)
+    {
+        if (move == MOVE_IVY_CUDGEL && (species == SPECIES_OGERPON_WELLSPRING_MASK || species == SPECIES_OGERPON_WELLSPRING_MASK_TERA
+                    || species == SPECIES_OGERPON_HEARTHFLAME_MASK || species == SPECIES_OGERPON_HEARTHFLAME_MASK_TERA
+                    || species == SPECIES_OGERPON_CORNERSTONE_MASK || species == SPECIES_OGERPON_CORNERSTONE_MASK_TERA))
+            type = type2;
+
+        else if (move == MOVE_STRUGGLE)
+            type = TYPE_NORMAL;
+
```

Comment:
is there a reason not to return early in all of those conditions? Also to stay consistent we use brackets everywhere if there are somewhere brackets in an if-else block and no new line inbetween

Reply:
some need not be returned early to account for Normalize i.e. Ivy Cudgel/Aura Wheel, although not possible to have that happen in vanilla it could be a case if abilities are changed. Same goes for Ion/Deluge/Electrify being over written by -Ate abilities. Setting up early returns on all that make no difference though